### PR TITLE
feat: allocate stack with best price

### PR DESCRIPTION
**Pull request by Cifko**
When no stacks are found for the given model. We find the best task with the node with the best price and allocate maximum number of tokens that the node allows us.

### CODE REVIEW

File: atoma-state/src/state_manager.rs [hunk 1]
```diff
@@ -284,6 +284,48 @@ impl AtomaState {
...
+    /// - `Result<Option<CheapestTask>>>`: A result containing either:
+    ///  - `Ok(Some<CheapestTask>)`: A `CheapestTask` object representing the task with the cheapest price.
+    ///  - `Ok(None)`: If no task is found for the given model.
+    ///  - `Err(AtomaStateManagerError)`: An error if the database query fails or if there's an issue parsing the results.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the database query fails.
+    #[instrument(level = "trace", skip_all, fields(%model))]
+    pub async fn get_cheapest_task_for_model(&self, model: &str) -> Result<Option<CheapestTask>> {
```
Comment (jorgeantonio21) [inline atoma-state/src/state_manager.rs:308]: This naming is erroneous. You should pick the cheapest stack currently available, with a given selected node. A task is not price bound.

**Review by jorgeantonio21 — COMMENTED** [review]
We need to pick the best node, based on price (other metrics in the future), amon the current available `Stack`s that are bound to a `Task` holding a specific model.

File: atoma-state/src/state_manager.rs (hunk 1 above)
Comment (Cifko) [reply atoma-state/src/state_manager.rs:308]: renamed to `get_cheapest_node_for_model`

File: atoma-state/src/handlers.rs [hunk 2]
```diff
@@ -719,6 +719,24 @@ pub(crate) async fn handle_state_manager_event(
...
+            model,
+            result_sender,
+        } => {
+            trace!(
+                target = "atoma-state-handlers",
+                event = "handle-state-manager-event",
+                "Getting cheapest node for model: {}",
+                model
+            );
+            let task = state_manager
```
Comment (jorgeantonio21) [inline atoma-state/src/handlers.rs:732]: maybe change this to `let node =`

File: atoma-state/src/state_manager.rs [hunk 3]
```diff
@@ -284,6 +284,50 @@ impl AtomaState {
...
+                  WHERE is_deprecated=false
+                    AND model_name = $1) AS tasks 
+            JOIN (SELECT * 
+                  FROM node_subscriptions
+                  WHERE valid = true) AS node_subscriptions 
+            ON tasks.task_small_id=node_subscriptions.task_small_id 
+            ORDER BY node_subscriptions.price_per_compute_unit 
+            LIMIT 1",
+        )
+        .bind(model)
```
Comment (jorgeantonio21) [inline atoma-state/src/state_manager.rs:322]: tab. ?

**Review by jorgeantonio21 — APPROVED** [review]
LGTM,

Left minor comments, that can be quickly addressed

File: atoma-state/src/state_manager.rs (hunk 3 above)
Comment (Cifko) [reply atoma-state/src/state_manager.rs:322]: this is the cargo fmt standard

**Pull request merged**